sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/functions.scala

```diff
@@ -256,12 +256,16 @@ case class Sum(child: Expression) extends AlgebraicAggregate {
   override def dataType: DataType = resultType
 
   // Expected input data type.
+  // TODO: Once we remove the old code path, we can use our analyzer to cast NullType
```

Comment:
We replace old aggregate functions to new aggregate functions at planning time. So, we need to have `NullType` at here to make this expression resolved.